src/lib/process.pl

```diff
@@ -0,0 +1,202 @@
+            ([stderr], valid_stdio, stderr(std), stderr(Stderr)),
+            ([env, environment], valid_env, environment([]), Env),
+            ([process], valid_uninit_process, process(_), process(Process)),
+            ([cwd], valid_cwd, cwd("."), cwd(Cwd))
+        ],
+        Options
+    ),
+    Stdin =.. Stdin1,
+    Stdout =.. Stdout1,
+    Stderr =.. Stderr1,
+    simplify_env(Env, Env1),
+    '$process_create'(Exe, Args, Stdin1, Stdout1, Stderr1, Env1, Cwd, Process).
+
+%% process_id(+Process, -Pid).
+%
+process_id(Process, Pid) :-
+    valid_process(Process, process_id/2),
+    write(valid), nl,
+    must_be(var, Pid),
+    write(var), nl, 
+    '$process_id'(Process, Pid).
+
+%% process_wait(+Process, Status).
+%
+% See `process_create/3` with `Options = []`
+%
+process_wait(Process, Status) :- process_wait(Process, Status, []).
+
+
+%% process_wait(+Process, Status, Options).
+%
+% Wait for the process behind the process handle `Process` to exit.
+%
+% When the process exits regulary `Status` will be unified with `exit(Exit)` where `Exit` is the processes exit code.
+% When the process exits was killed `Status` will be unified with `killed(Signal)` where `Signal` is the signal number that killed the process.
+% When the process doesn't exit before the timeout `Status` will be unified with `timeout`.
+%
+% `Options` is a a list of the following options
+%
+%  * timeout(Timeout) supported values for `Timeout` are 0 or `infinite`
+%
+% Each options may be specified at most once, when an option is not specified the following defaults apply:
+%
+% - timeout(infinite)
+%
+process_wait(Process, Status, Options) :- 
+    valid_process(Process, process_wait/3),
+    must_be_known_options([timeout], [], Options),check_options(
+        [
+            ([timeout], valid_timeout, timeout(infinite), timeout(Timeout))
+        ],
+        Options
+    ),
+    '$process_wait'(Process, Exit, Timeout),
+    Exit = Status.
+
+valid_timeout(timeout(infinite)).
+valid_timeout(timeout(0)).
+
+
+%% process_kill(+Process).
+%
+% Kill the process using the process handle `Process`.
+% On Unix this sends SIGKILL.
+%
+% Only works for processes spawned with `process_create/3` that have not yet been release with `process_release/1`
+% 
+process_kill(Process) :- 
+    valid_process(Process, process_kill/1),
+    '$process_kill'(Process).
+
+%% process_release(+Process)
+%
+% wait for the process to exit (if not already) and release process handle `Process`
+%
+% It's an error if `Process` is not a valid process handle
+%
+process_release(Process) :- 
+    valid_process(Process, process_release/1),
+    process_wait(Process, _),
+    '$process_release'(Process).
+
+
+must_be_known_options(_, _,  []).
+must_be_known_options(Valid, Found, [X|XS]) :-
+    X =.. [Option|_],
+    (
+        member(Option, Found) -> domain_error(non_duplicate_process_create_options, process_create/3);
+        member(Option, Valid) -> true ;
```

Comment:
Please reformat so that `;` never appears at the *end* of a line, where it will inevitably be misread as&nbsp;`,`.